src/pages/workspace/WorkspaceSettingsPage.js

```diff
@@ -0,0 +1,221 @@
+};
+const defaultProps = {
+    betas: [],
+};
+
+class WorkspaceSettingsPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            name: props.policy.name,
+            avatarURL: props.policy.avatarURL,
+            previewAvatarURL: props.policy.avatarURL,
+            currency: props.policy.outputCurrency,
+        };
+
+        this.submit = this.submit.bind(this);
+        this.uploadAvatar = this.uploadAvatar.bind(this);
+        this.removeAvatar = this.removeAvatar.bind(this);
+        this.getCurrencyItems = this.getCurrencyItems.bind(this);
+        this.uploadAvatarPromise = Promise.resolve();
+    }
+
+    componentDidMount() {
+        getCurrencyList();
+    }
+
+    /**
+     * @returns {Object[]}
+     */
+    getCurrencyItems() {
+        const currencyListKeys = _.keys(this.props.currencyList);
+        return _.map(currencyListKeys, currencyCode => ({
+            value: currencyCode,
+            label: `${currencyCode} - ${this.props.currencyList[currencyCode].symbol}`,
+        }));
+    }
+
+    removeAvatar() {
+        this.setState({previewAvatarURL: '', avatarURL: ''});
+    }
+
+    /**
+     * @param {Object} image
+     * @param {String} image.uri
+     */
+    uploadAvatar(image) {
+        Policy.updateLocalPolicyValues(this.props.policy.id, {isAvatarUploading: true});
+        this.setState({previewAvatarURL: image.uri});
+
+        // Store the upload avatar promise so we can wait for it to finish before updating the policy
+        this.uploadAvatarPromise = Policy.uploadAvatar(image).then(url => new Promise((resolve) => {
+            this.setState({avatarURL: url}, resolve);
+        })).catch(() => {
+            Growl.error(this.props.translate('workspace.editor.avatarUploadFailureMessage'));
+        }).finally(() => Policy.updateLocalPolicyValues(this.props.policy.id, {isAvatarUploading: false}));
+    }
+
+    submit() {
+        Policy.updateLocalPolicyValues(this.props.policy.id, {isPolicyUpdating: true});
+
+        // Wait for the upload avatar promise to finish before updating the policy
+        this.uploadAvatarPromise.then(() => {
+            const name = this.state.name.trim();
+            const avatarURL = this.state.avatarURL;
+            const policyID = this.props.policy.id;
+            const currency = this.state.currency;
+
+            Policy.update(policyID, {name, avatarURL, outputCurrency: currency});
+        }).catch(() => {
+            Policy.updateLocalPolicyValues(this.props.policy.id, {isPolicyUpdating: false});
+        });
+    }
+
+    render() {
+        const {policy} = this.props;
+
+        if (!Permissions.canUseFreePlan(this.props.betas)) {
+            console.debug('Not showing workspace editor page because user is not on free plan beta');
+            return <Navigation.DismissModal />;
+        }
+
+        if (_.isEmpty(policy)) {
+            return <FullScreenLoadingIndicator visible />;
```

Comment:
NAB but visible is the default
```suggestion
            return <FullScreenLoadingIndicator />;
```